pkg/rbac/parser_integration_test.go

```diff
@@ -43,19 +45,29 @@ var _ = Describe("ClusterRole generated by the RBAC Generator", func() {
 			}
 
 			By("generating a ClusterRole")
-			clusterRole, err := rbac.GenerateClusterRole(ctx, "manager-role")
+			objs, err := rbac.GenerateRoles(ctx, "manager-role")
 			Expect(err).NotTo(HaveOccurred())
 
 			By("loading the desired YAML")
 			expectedFile, err := ioutil.ReadFile("role.yaml")
 			Expect(err).NotTo(HaveOccurred())
 
 			By("parsing the desired YAML")
-			var role rbacv1.ClusterRole
-			Expect(yaml.Unmarshal(expectedFile, &role)).To(Succeed())
+			for i, expectedRoleBytes := range bytes.Split(expectedFile, []byte("\n---\n"))[1:] {
```

Comment:
It's a little too complex for a simple unit test.
Since we have sorted everything. We can just compare the 2 objects in serialized format, can't we? We return the 1st place that mismatches.

Reply:
`rbac.GenerateRoles` generate a list of objs, to serialize these objects into a single file, we need to combine all the objects and  add line break `\n---\n` between objects. This does not seem to much simpler than the existing code in this PR. Another advantage of the existing code is that we can give users more specific diff.